specification/_types/query_dsl/specialized.ts

```diff
@@ -201,3 +202,13 @@ export class ShapeFieldQuery {
   relation?: GeoShapeRelation
   shape?: GeoShape
 }
+
+/**
+ * @variants container
+ */
+export class RuleQuery extends QueryBase {
+  /** @variant container_property */
+  organic: QueryContainer
+  ruleset_id?: QueryRulesetId
```

Comment:
Are `ruleset_id` and `match_criteria` optional?

Reply:
They shouldn't be, but making them required results in the following error from `make generate`: 

```
Error: All @variants container properties must be optional
```

Reply:
Got it, thanks! I think that `ruleset_id` can be tagged as `container_property` in that case according to [this example](https://github.com/elastic/elasticsearch-specification/blob/5ccd90cfd3030b32527928484bfc255746124a3f/docs/modeling-guide.md?plain=1#L399).

Reply:
Fixed, thanks!